src/models/kinflated_gamma_poisson_model.py

```diff
@@ -0,0 +1,633 @@
+        _exponential_poisson_reach for beta in terms of I, N and R.
+
+        Args:
+          I:  The hypothetical number of impressions purchased.
+          N:  The audience size.
+          R:  The 1+ reach that was obtained.
+        Returns:
+          The corresponding parameter beta of the exponential-poisson
+          model that achieves reach R when I impressions are delivered
+          to an audience of size N.
+        """
+        if I * R - N * (I - R) <= 0:
+            return 1e3
+        return (N * (I - R)) / (I * R - N * (I - R))
+
+    def _exponential_poisson_N_from_beta(
+        self, I: float, R: float, beta: float
+    ) -> float:
+        """Estimate audience size from impressions, reach and beta.
+
+        This is computed by backsolving the formula for reach used in
+        _exponential_poisson_reach for N in terms of I, R and beta.
+
+        Args:
+          I:  The hypothetical number of impressions purchased.
+          R:  The 1+ reach that was obtained.
+          beta: The parameter of the exponential-poisson distribution.
+        Returns:
+          The corresponding parameter N of the exponential-poisson
+          model that achieves reach R when I impressions are delivered
+          to an audience of size N with parameter beta.
+        """
+        if I < R + 1e-3:
+            return 100.0 * R
+        return -(I * beta * R) / ((beta + 1) * (R - I))
+
+    def _exponential_poisson_N(self, I: float, R: float, mu_scale_factor=1.5) -> float:
+        """Estimate audience size from I and R.
+
+        We know that the mean mu of the distribution has to be greater than
+        I / R.  Since beta = mu - 1, this says that beta >= I / R - 1.
+        But how much bigger?  That cannot easily be determined from just one
+        point.  Here, we assume beta = s * (I / R - 1), where s is some
+        small scale factor that is at least one.  From that, we estimate N.
+        """
+        if mu_scale_factor < 1.0:
+            raise ValueError("mu_scale_factor must be at least 1.0")
+        elif I < R + 1e-3:
+            return 100.0 * R
+        return R * (mu_scale_factor * I - R) / (mu_scale_factor * (I - R))
+
+    def _fit_exponential_poisson_model(
+        self, point: ReachPoint, Imin: float = None
+    ) -> Tuple[float, KInflatedGammaPoissonDistribution]:
+        """Returns N, alpha, beta of an exponential-poisson model.
+
+        The fit returned by this function is guaranteed to match the 1+ reach
+        of the input point.  If additional frequencies are given, there should
+        be a reasonable match at these as well, although the match is not
+        guaranteed to be perfect.
+
+        The parameters returned by this function are used to bootstrap the
+        full k-Inflated Gamma Poisson model.
+
+        Args:
+          point:  A ReachPoint to which an exponential-poisson model is
+            to be fit.
+          Imin:  Minimum number of impressions that the returned model
```

Comment:
Minimum expected number of inventory impressions

Reply:
Done.